src/lib/libgit2/note.zig

```diff
@@ -43,27 +44,27 @@ pub const GitNote = struct {
         if (res < 0) {
             if (res == c.GIT_EEXISTS) {
                 // Note already exists, update it instead
-                return update(repo, commit_id, note_content, signature);
+                return update(repo, commit_id, note_content, signature, notes_ref);
             }
             log.err("Failed to create note with error code: {d}", .{res});
             return GitError.NOTE_CREATE_FAILED;
         }
```

Comment:
Just realized that making libgit2 wrapper smarter is not really needed and shouldnt be done at all. So updating a note if it exist already shouldnt be concern of libgit2 implementation. This should be handled in higher layers in our case is should be handled in sparse side instead of libgit2